pandas/_libs/lib.pyx

```diff
@@ -1209,6 +1215,10 @@ def infer_dtype(value: object, skipna: bool=False) -> str:
         values = construct_1d_object_array_from_listlike(value)
 
     values = getattr(values, 'values', values)
+
```

Comment:
Because things like `tests.dtypes.test_inference::TestTypeInference::test_length_zero` would fail otherwise.

It's a bug that was uncaught up until now that
```
arr = np.array([np.array([], dtype=object),
                np.array([], dtype=object)])
result = lib.infer_dtype(arr, skipna=True)
```

couldn't work, because it tries:

```
if skipna:
    values = values[~isnaobj(values)]
```
on a 2-dim object.

Moving the `.ravel` before that fixes the issue.